core/src/main/java/io/crate/types/DataType.java

```diff
@@ -101,6 +104,14 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
no, i followed the pattern we use for streaming in this class. As only Array and Set need to do additional stuff it would be annoying to enforce implementation